src/NRules/NRules.Testing/IRepositorySetup.cs

```diff
@@ -0,0 +1,11 @@
+﻿using NRules.Fluent;
+using NRules.Fluent.Dsl;
+
+namespace NRules.Testing;
+
+public interface IRepositorySetup
+{
+    void Rule<T>() where T : Rule;
+
+    void Rule(IRuleMetadata metadata);
```

Comment:
@gritcsenko I don't think you need an overload with IRuleMetadata. You just need a non-generic Rule(Type ruleType) and instantiate the metadata inside.

Reply:
Updated, thanks